play/src/main/scala/org/lyranthe/prometheus/client/play/filters/PrometheusFilter.scala

```diff
@@ -0,0 +1,57 @@
+package org.lyranthe.prometheus.client.play.filters
+
+import org.lyranthe.prometheus.client.{play => _, _}
+import org.lyranthe.prometheus.client.histogram.LabelledHistogram
+import play.api.mvc._
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success, Try}
+
+class PrometheusFilter()(implicit val registry: Registry, executionContext: ExecutionContext) extends Filter {
+
+  private val ClientErrorClass = "4xx"
+
+  private val ServerErrorClass = "5xx"
+
+  private val RouteRegex = "[/a-zA-Z0-9$_\\-]+".r
+
+  private val httpHistogramBuckets = {
+    val buckets = for(p <- Vector[Double](0.0001, 0.001, 0.01, 0.1, 1, 10); s <- Vector(1, 2, 5)) yield (p * s)
+    HistogramBuckets(buckets: _*)
+  }
+
+
+  private val httpRequestLatency =
+    Histogram(metric"http_request_duration_seconds", "Duration of HTTP request in seconds")(httpHistogramBuckets)
+      .labels(label"path", label"status")
+      .register
+
+
+  private val counter = Counter(metric"http_request_route_not_found", "Route not found")
```

Comment:
Following https://prometheus.io/docs/practices/naming/ maybe call this:
`http_request_badregex_total`

(in particular, `_total` refers to a unit-less accumulating counter). And I think this metric name is closer to explaining what's actually happening.